server/routes/metrics/metrics_rounter.ts

```diff
@@ -62,4 +70,153 @@ export function registerMetricsRoute(router: IRouter) {
       }
     }
   );
+
+  router.get(
+    {
+      path: `${OBSERVABILITY_BASE}/search/indices`,
+      validate: {},
+    },
+    async (context, request, response) => {
+      const indexPattern = 'ss4o_metrics-*-*';
+      const params = {
+        format: 'json',
+        index: indexPattern,
+      };
+      try {
+        const resp = await context.core.opensearch.legacy.client.callAsCurrentUser(
+          'cat.indices',
+          params
+        );
+        return response.ok({
+          body: resp,
+        });
+      } catch (error) {
+        if (error.statusCode !== 404) console.error(error);
+        return response.custom({
+          statusCode: error.statusCode || 500,
+          body: error.message,
+        });
+      }
+    }
+  );
+
+  router.post(
+    {
+      path: `${OBSERVABILITY_BASE}/metrics/otel/documentNames`,
+      validate: {
+        body: schema.object({
+          index: schema.string(),
+        }),
+      },
+    },
+    async (
+      context,
+      request,
+      response
+    ): Promise<IOpenSearchDashboardsResponse<any | ResponseError>> => {
+      const opensearchNotebooksClient: ILegacyScopedClusterClient = context.observability_plugin.observabilityClient.asScoped(
+        request
+      );
+
+      try {
+        const resp = await metricsAnalyticsBackend.queryToFetchDocumentNames(
+          opensearchNotebooksClient,
+          request.body.index
+        );
+        return response.ok({
+          body: resp,
+        });
+      } catch (error) {
+        if (error.statusCode !== 404) console.error(error);
+        return response.custom({
+          statusCode: error.statusCode || 500,
+          body: error.message,
+        });
+      }
+    }
+  );
+
+  router.post(
+    {
+      path: `${OBSERVABILITY_BASE}/metrics/otel/sampleDocument`,
```

Comment:
Question: What is `sampleDocument` endpoint used for? Is it for samples?

Reply:
Each document variety will have different bin ranges, we need bin ranges for the next query to fetch count so this API call is used to retrieve one of the document to find the bin ranges

Reply:
Suggestion: Do you want to change the path url to be something more meaningful. Moreover, since this is not performing any inserts/updates to the backend, a fetch request can map to `GET` request. Here's a good write up on how to create API endpoints: https://medium.com/fullstacked/restful-api-cheat-sheet-3f96fab970b8

Reply:
Are you planning to change this to a `get` call?

Reply:
I can't change it into `get` call as it doesn't have an option to send query in body. Only POST request can be used to send the json query which contains `documentName`

Reply:
why is a json query being sent for `documentName`. Shouldn't that just be a string?

Reply:
no, its actually a json query. Please refer this - https://github.com/kavithacm/dashboards-observability/blob/data-prepper-metrics/server/adaptors/metrics/metrics_analytics_adaptor.ts#L89

Reply:
from the above link  do see docname and index are strings: https://github.com/kavithacm/dashboards-observability/blob/4d11381ec51e28c43bd7f410f2580c7e76fc8978/server/adaptors/metrics/metrics_analytics_adaptor.ts#L91-L92. The json is being sent only to OpenSearch backend. Whereas, the node API still just needs two string values.